sentry/silkworm/sentry/rlpx/auth/ecies_cipher.cpp

```diff
@@ -68,29 +75,40 @@ EciesCipher::Message EciesCipher::encrypt_message(ByteView plain_text, PublicKey
     };
 }
 
-Bytes EciesCipher::decrypt_message(const EciesCipher::Message& message, PrivateKeyView private_key) {
-    secp256k1_pubkey ephemeral_public_key;
-    assert(message.ephemeral_public_key.size() == sizeof(ephemeral_public_key.data));
-    memcpy(ephemeral_public_key.data, message.ephemeral_public_key.data(), sizeof(ephemeral_public_key.data));
-
-    Bytes shared_secret(kKeySize * 2, 0);
-    SecP256K1Context ctx;
-    bool ok = ctx.compute_ecdh_secret(shared_secret, &ephemeral_public_key, private_key);
-    if (!ok) {
-        throw std::runtime_error("Failed to ECDH-agree private key and ephemeral public key");
-    }
-
+Bytes EciesCipher::decrypt_message(const EciesCipher::Message& message, PrivateKeyView private_key, ByteView mac_extra_data) {
+    Bytes shared_secret = kdf(compute_shared_secret(message.ephemeral_public_key, private_key));
     ByteView aes_key(shared_secret.data(), kKeySize);
     ByteView mac_key(&shared_secret[kKeySize], kKeySize);
 
-    Bytes mac = hmac(sha256(mac_key), message.iv, message.cipher_text);
+    Bytes mac = hmac(sha256(mac_key), message.iv, message.cipher_text, mac_extra_data);
     if (mac != message.mac) {
         throw std::runtime_error("Invalid MAC");
     }
 
     return aes_decrypt(message.cipher_text, aes_key, message.iv);
 }
 
+size_t EciesCipher::round_up_to_block_size(size_t size) {
+    return (size + AES_BLOCK_SIZE - 1) / AES_BLOCK_SIZE * AES_BLOCK_SIZE;
+}
+
+size_t EciesCipher::estimate_encrypted_size(size_t size) {
+    return size
+        + SecP256K1Context::kPublicKeySizeUncompressed
+        + AES_BLOCK_SIZE
+        + kMacSize;
+}
+
+// NIST SP 800-56 Concatenation Key Derivation Function (see section 5.8.1).
+// Since sha256 produces the right size, one iteration is enough.
+static Bytes kdf(ByteView secret) {
+    assert(secret.size() == kKeySize * 2);
```

Comment:
I suggest using `SILKWORM_ASSERT` instead of `assert` for uniformity w/ the rest of the code base

Reply:
Should I change everywhere in my code then?

I've used `assert()` as a sanity check while debugging in places where I think it'd be unexpected given the assumptions around the code, and thus redundant to check in Release mode. If I need to check in Release, I'd rather throw an exception than crash.

Could we make SILKWORM_ASSERT debug-only by default (and e.g. have a build flag for enabling it in Release)?